spring-xd-dirt/src/main/java/org/springframework/xd/dirt/cluster/ContainerFilter.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.dirt.cluster;
+
+import java.util.List;
+
+import org.springframework.xd.module.ModuleDescriptor;
+
+/**
+ * A filter interface that applies filter for the given containers and returns
+ * eligible containers for deployment.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public interface ContainerFilter {
```

Comment:
After discussion with @pperalta this is the proposed plan:

1. merge this PR without changing anything related to ContainerMatcher or ContainerFilter yet
2. refactor RedploymentContainerMatcher so that it implements ContainerFilter
3. remove the ContainerMatcher interface and rename DefaultContainerMatcher to just ContainerMatcher
4. move ContainerMatcher to the .server package *and* make it non-public